yorm/utilities.py

```diff
@@ -37,10 +41,81 @@ def find(class_or_instance, *args, create=False, **kwargs):  # pylint: disable=r
         return None
 
 
-def match(cls, **kwargs):
-    """Yield all matching mapped objects."""
-    log.debug((cls, kwargs))
-    raise NotImplementedError
+class GlobFormatter(string.Formatter):
+    """
+    Uses '*' for all unknown fields
+    """
+
+    WILDCARD = object()
+
+    def get_field(self, field_name, args, kwargs):
+        try:
+            return super().get_field(field_name, args, kwargs)
+        except (KeyError, IndexError, AttributeError):
+            return self.WILDCARD, None
+
+    def get_value(self, key, args, kwargs):
+        try:
+            return super().get_value(key, args, kwargs)
+        except (KeyError, IndexError, AttributeError):
+            return self.WILDCARD
+
+    def convert_field(self, value, conversion):
+        if value is self.WILDCARD:
+            return self.WILDCARD
+        else:
+            return super().convert_field(value, conversion)
+
+    def format_field(self, value, format_spec):
+        if value is self.WILDCARD:
+            return '*'
+        else:
+            return super().format_field(value, format_spec)
+
+
+def _unpack_parsed_fields(pathfields):
+    return {
+        (k[len('self.'):] if k.startswith('self.') else k): v
+        for k, v in pathfields.items()
+    }
+
+
+def match(cls_or_path, factory=None, **kwargs):
```

Comment:
-1, +1

I think it should remain a positional argument to avoid conflicting with fields, which means it should have a name to avoid conflicting with fields.